mlflow/tracing/types/wrapper.py

```diff
@@ -1,32 +1,89 @@
 import json
 import logging
-from functools import lru_cache
-from typing import Any, Dict, Optional, Union
-
-from opentelemetry import trace as trace_api
-
-from mlflow.entities import Span, SpanContext, SpanEvent, SpanStatus, SpanType, TraceStatus
+from dataclasses import asdict
+from functools import lru_cache, wraps
+from typing import Any, Dict, List, Optional, Union
+
+from opentelemetry.sdk.trace import Event as OTelEvent
+from opentelemetry.sdk.trace import ReadableSpan as OTelReadableSpan
+from opentelemetry.trace import Span as OTelSpan
+
+from mlflow.entities.span import SpanType
+from mlflow.entities.span_event import SpanEvent
+from mlflow.entities.span_status import SpanStatus
+from mlflow.entities.trace_status import TraceStatus
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
 from mlflow.tracing.types.constant import SpanAttributeKey
-from mlflow.tracing.utils import TraceJSONEncoder, format_span_id, format_trace_id
+from mlflow.tracing.utils import (
+    TraceJSONEncoder,
+    build_otel_context,
+    decode_span_id,
+    decode_trace_id,
+    format_span_id,
+    format_trace_id,
+)
 
 _logger = logging.getLogger(__name__)
 
 
-class MlflowSpanWrapper:
+def live_span_only(func):
     """
-    A wrapper around OpenTelemetry's Span object to provide MLflow-specific functionality.
+    A decorator to ensure that the method is called only on "live"" spans. A live span is a span
+    that is created and updated during the application runtime, in contrast to the spans that are
+    loaded back from the storage. The latter is immutable so setter methods should not be called.
+    """
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        self = args[0]
```

Comment:
```suggestion
    def wrapper(self, *args, **kwargs):
```

if we only use this decorator for bound methods.

Reply:
Good idea!